src/OpenSSL/SSL.py

```diff
@@ -808,20 +809,22 @@ def set_tmp_ecdh(self, curve):
 
     def set_cipher_list(self, cipher_list):
         """
-        Change the cipher list
+        Set the list of ciphers to be used in this context.
 
-        :param cipher_list: A cipher list, see ciphers(1)
+        See the OpenSSL manual for more information (e.g.
+        :manpage:`ciphers(1)`).
+
+        :param bytes cipher_list: An OpenSSL cipher string.
         :return: None
         """
-        if isinstance(cipher_list, _text_type):
-            cipher_list = cipher_list.encode("ascii")
+        cipher_list = _text_to_bytes_and_warn("cipher_list", cipher_list)
```

Comment:
Well usually we don’t check the types of our arguments since a TypeError would bubble up from CFFI anyway.  I guess pyOpenSSL is full of things that only make sense through the C-API lens.